### PR TITLE
release tools

### PR DESCRIPTION
**Pull request by pohly**
This syncs the build rules with the other kubernetes-csi repos.

### CODE REVIEW

**Comment by pohly** [comment]
/hold
Need to update the PR with a fix for publishing "canary" once merged into master (https://github.com/kubernetes-csi/csi-release-tools/pull/4) and also get the local `release--tools` changes upstream (https://github.com/kubernetes-csi/csi-release-tools/pull/5).

**Comment by pohly** [comment]
Rebased because of the merge conflict with https://github.com/kubernetes-csi/csi-test/commit/4df3edebbe498cc3f76e849017245af686b2788e

**Comment by pohly** [comment]
The new check actually works as intended:
```
### test-vet:
go vet `go list ./... | grep -v vendor `
# github.com/kubernetes-csi/csi-test/mock/service
mock/service/controller.go:511: Entry.Infof format %s has arg requestBytes of wrong type int64
make: *** [test-vet] Error 2
```

I need to fix the code before enforcing this check.

**Comment by davidz627** [comment]
Hi @pohly quick qol request from me. I'm wondering if you could add the `[WIP]` tag to your PR's when they are still works in progress and not ready for review. That would be very helpful for reviewers trying to figure out which PR's to look at and which can be reasonably skipped for now.

**Comment by pohly** [comment]
PR #166 has to be merged first to get a clean test result.

File: release-tools/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+[`git subtree`](https://github.com/git/git/blob/master/contrib/subtree/git-subtree.txt)
+is the recommended way of maintaining a copy of the rules inside the
+`release-tools` directory of a project. This way, it is possible to make
+changes also locally, test them and then push them back to the shared
+repository at a later time.
+
+Cheat sheet:
+
+- `git subtree add --prefix=release-tools https://github.com/kubernetes-csi/csi-release-tools.git master` - add release tools to a repo which does not have them yet (only once)
```
Comment (lpabon) [inline release-tools/README.md]: I have to be honest about this, I feel this is really overkill for this project and is quite different from a normal git release process.

I'm concerned that I am going to mess something up (that's my biggest concern)

File: release-tools/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Release Process
+
+No tagged releases are planned at this point. The intention is to keep
```
Comment (lpabon) [inline release-tools/RELEASE.md]: Should this have the explanation on how to use this, or is it the README.md that we should follow?

File: release-tools/build.make [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+container-%: build-%
+	docker build -t $*:latest -f $(shell if [ -e ./cmd/$*/Dockerfile ]; then echo ./cmd/$*/Dockerfile; else echo Dockerfile; fi) --label revision=$(REV) .
+
+push-%: container-%
+	set -ex; \
+	push_image () { \
+		docker tag $*:latest $(IMAGE_NAME):$$tag; \
+		docker push $(IMAGE_NAME):$$tag; \
+	}; \
+	for tag in $(IMAGE_TAGS); do \
```
Comment (lpabon) [inline release-tools/build.make]: To me this is almost magic. I am concerned with the maintenance of these tools and scripts. I feel only the author can maintain it and that is concerning.

**Comment by lpabon** [comment]
After an offline discussion on `#wg-csi` @pohly and @msau42 will maintain this code.

**Pull request merged**